CMake/IBMCompilers.cmake

```diff
@@ -54,3 +55,43 @@ IF(QMC_BITS MATCHES 64)
         "<CMAKE_AR> -X64 cr <TARGET> <LINK_FLAGS> <OBJECTS> " 
         "<CMAKE_RANLIB> <TARGET> ")
 ENDIF(QMC_BITS MATCHES 64)
+ENDIF(CMAKE_SYSTEM_NAME MATCHES "AIX")
+
+IF(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+MESSAGE(STATUS "Power8+ system using xlC/xlc/xlf")
+
+ADD_DEFINITIONS( -Drestrict=__restrict__ )
+ADD_DEFINITIONS( -DADD_ )
+ADD_DEFINITIONS( -DINLINE_ALL=inline )
+
+# Clean up flags
+
+IF(CMAKE_C_FLAGS MATCHES "-qhalt=e")
+  SET(CMAKE_C_FLAGS "")
+ENDIF()
+IF(CMAKE_CXX_FLAGS MATCHES "-qhalt=e")
+  SET(CMAKE_CXX_FLAGS "")
+ENDIF()
+
+# Suppress compile warnings
+SET(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -Wno-deprecated -Wno-unused-value")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated -Wno-unused-value")
+
+# Set standard and other optimization settings
+SET(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -std=gnu11 -O3")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -qnoxlcompatmacros -O3")
+
```

Comment:
Both C and CXX compiler flags going into IBMCompilers.cmake are "-qthreaded -qhalt=e".